src/test/java/org/openmrs/contrib/qaframework/automation/RegistrationSteps.java

```diff
@@ -12,16 +11,18 @@
 import org.openmrs.reference.page.RegistrationPage;
 import org.openqa.selenium.By;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
+import io.cucumber.java.After;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
 
 public class RegistrationSteps extends Steps {
 
 	RegistrationPage registrationPage;
 	TestPatient patient;
 	ClinicianFacingPatientDashboardPage dashboardPage;
 
```

Comment:
I expect @Before method some where since  `RegistrationSteps` is being detached from the orders dwelling in `qa.yml` to rather use orders for `registration.yml`

Reply:
Lets  not add before method because its not by default that we need before method every implementation. In before method, i would add a functionality of  initalizing loginPage  , However this is being capture behind the scenes

Reply:
> Lets not add before

Which logic will be responsible for the test login into the system ? And what will be the background of the test ?

Reply:
By default goToHomePage will irrespective of adding a before method. Am also following the initial implementation

Reply:
Now that we making the workflow to be independent so that we track its status on the Dashboard, we need to have its own logic that logs the test into the system and this can be achieved within @Before method.

> Am also following the initial implementation.

The initial implementation of the test was being governed by commands that are dwelling in `qa.yml,`  you need to make sure that  the workflow is governed by the commands from `registration.yml`

Reply:
I already have an implementation written for [registration.yml](https://github.com/openmrs/openmrs-contrib-qaframework/pull/102/files)

Reply:
Another suggestion why we might not need before hook method is that we already have an implementation of login in[ here ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/102/files#diff-68d8685a62f4a01878be56c191b10ca50a9a679b8d3047f0eedc2265d5370f79R32), so using[ initiateLogin()](https://github.com/openmrs/openmrs-contrib-qaframework/pull/101/files#diff-d7d118d304f2585ada64c935320d5efb7ca710d547dfb7aa45bc5b89165b073cR23) will throw duplicate exception.